src/config/resolver.rs

```diff
@@ -0,0 +1,48 @@
+use serde::{Deserialize, Serialize};
+use url::Url;
+use crate::config::is_default;
+use anyhow::anyhow;
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+
+#[derive(Default, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+enum LinkType {
+  #[default]
+  Config,
+  GraphQL,
+  Protobuf,
+  Data,
+}
+
+#[derive(Default, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct Link {
+  #[serde(default, skip_serializing_if = "is_default")]
+  type_of: LinkType, // Type of the link
+  #[serde(default, skip_serializing_if = "is_default")]
+  src: String, // Source URL for linked files
+  #[serde(default, skip_serializing_if = "is_default")]
+  id: Option<String>, // Id is used to refer at different places in the config
+  #[serde(default, skip_serializing_if = "is_default")]
+  content: Option<String>, // Stores raw content
+}
+
+impl Link {
+
+  pub async fn read_link_type(&mut self) -> Result<(), anyhow::Error> {
+    if let Ok(url) = Url::parse(&self.src) {
+      let resp = reqwest::get(url).await?;
+      if !resp.status().is_success() {
+        return Err(anyhow!("Read over URL failed with status code: {}", resp.status()));
+      }
+      self.content = Some(resp.text().await?);
+    } else {
+      let path = &self.src.trim_end_matches('/');
+      let mut f = File::open(path).await?;
+      let mut buffer = Vec::new();
+      f.read_to_end(&mut buffer).await?;
+      self.content = Some(String::from_utf8(buffer)?);
+    };
+    
+    Ok(())
```

Comment:
Need to check for recursive links so that it doesn't get into a loop. 

- [ ] Add unit tests for this component
- [ ] Keep File and Http as a dependency so that it's easier to test.

Reply:
@tusharmath Regarding the recursive links, the `resolver` module only fetches the file content and places it into the `content` attribute of the `Link`, as it was mentioned that 
`Focused Module: The ConfigResolver is dedicated to fetching and aggregating content. Content parsing is handled in the config-to-blueprint step, ensuring modularity.`
Since content parsing is not done in this module It doesn't know if the linked file/URL contains any additional links. If we consider this case, we need to parse the content and add the links to the `Vec<Link>`.

Reply:
`blueprint::from_config` is not async. So we can't perform any IOs in that step. Everything will need to be resolved before it reaches that step. We could parse the files and insert it in the config as data but not validate it. Parsing could be the responsibility of the Resolver module.

Reply:
adding the unit test in the next commit :+1: